src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -49,28 +62,16 @@ internal sealed partial class LocalRewriter
 
             statementList.AddRange(existingStatements);
             return statementList.ToImmutableAndFree();
-
         }
 
-        private static BoundStatement ConstructIfStatementForParameter(ParameterSymbol parameter, SyntheticBoundNodeFactory factory)
+        private static BoundStatement ConstructNullCheck(ParameterSymbol parameter, MethodSymbol throwIfNullMethod, SyntheticBoundNodeFactory factory)
         {
-            BoundExpression paramIsNullCondition;
-            var loweredLeft = factory.Parameter(parameter);
-
-            if (loweredLeft.Type.IsNullableType())
-            {
-                paramIsNullCondition = factory.Not(factory.MakeNullableHasValue(loweredLeft.Syntax, loweredLeft));
-            }
-            else
-            {
-                paramIsNullCondition = factory.MakeNullCheck(loweredLeft.Syntax, loweredLeft, BinaryOperatorKind.Equal);
-            }
-
-            var argumentName = ImmutableArray.Create<BoundExpression>(factory.StringLiteral(parameter.Name));
-            BoundObjectCreationExpression ex = factory.New(factory.WellKnownMethod(WellKnownMember.System_ArgumentNullException__ctorString), argumentName);
-            BoundThrowStatement throwArgNullStatement = factory.Throw(ex);
-
-            return factory.HiddenSequencePoint(factory.If(paramIsNullCondition, throwArgNullStatement));
+            var call = factory.Call(
+                receiver: null,
+                throwIfNullMethod,
+                arg0: factory.Convert(factory.SpecialType(SpecialType.System_Object), factory.Parameter(parameter)),
```

Comment:
Do we have tests for `void M(void* ptrParam!!)`?

Reply:
We don't.. but it is an error in the current implementation, and the [current proposal](https://github.com/dotnet/csharplang/blob/3a6c79012a8c53fe84330c09080105ac177ff93d/proposals/null-arg-checking.md) doesn't mention it.

It seems not-wrong to me to support pointers with the feature. We did after all even make `ptr is null` legal at some point. Perhaps an email to LDT would be reasonable and it's more motivation to restore the "non-helper" code path.

Reply:
Fixed by adding support for pointer types.